lib/AST/Decl.cpp

```diff
@@ -1467,20 +1467,31 @@ static Type mapSignatureFunctionType(ASTContext &ctx, Type type,
     SmallVector<TupleTypeElt, 4> elements;
     bool anyChanged = false;
     unsigned idx = 0;
+    // Remap our parameters, and make sure to strip off @escaping
     for (const auto &elt : tupleTy->getElements()) {
-      Type eltTy = mapSignatureParamType(ctx, elt.getType());
-      if (anyChanged || eltTy.getPointer() != elt.getType().getPointer()) {
-        if (!anyChanged) {
-          elements.append(tupleTy->getElements().begin(),
-                          tupleTy->getElements().begin() + idx);
-          anyChanged = true;
-        }
+      auto newEltTy = mapSignatureParamType(ctx, elt.getType());
+      ParameterTypeFlags newParamFlags(
+        elt.getParameterFlags().value & ~ParameterTypeFlags::Escaping);
```

Comment:
Yeah, I was debating between the two. Because I wanted this to feel more first class, that is have more friendly isFoo() methods and constructor, I sided with rolling my own. I'll try it out and see. Otherwise, I'll definitely make the other constructors private, etc.